src/averageAges.js

```diff
@@ -15,47 +15,62 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let males = people.filter(({ sex }) => sex === 'm');
+
+  if (century) {
+    males = males.filter(male => Math.ceil(male.died / 100) === century);
+  }
+
+  const averageAges = males.map(
+    male => {
+      return male.died - male.born;
+    })
+    .reduce(
+      (ageOfPreviousMale, ageOfCurrentMale) =>
+        ageOfPreviousMale + ageOfCurrentMale
+    ) / males.length;
```

Comment:
this reduce method as well can be used as a global variable
`const averageAgeCalculate = (ages) =>
  ages.reduce((sum, age) => sum + age, 0) / ages.length;`

Reply:
agree, let's fix it, you'll need the same function ahead in your code here